external-crates/move/crates/move-vm-runtime/src/jit/execution/translate.rs

```diff
@@ -1034,21 +1043,37 @@ fn flatten_and_renumber_blocks(
     dbg_println!("Concatenated: {:#?}", concatenated);
 
     // Rewrite branch instructions with new offsets
-    concatenated
+    let result = concatenated
         .into_iter()
         .map(|bytecode| match bytecode {
             input::Bytecode::BrFalse(target) => {
-                input::Bytecode::BrFalse(*offset_map.get(&target).expect("Invalid branch target"))
```

Comment:
This change is not necessary, because we would have already failed package verification before we got here.